### PR TITLE
dnscrypt-proxy service: auto-updated resolver list

### PR DESCRIPTION
**Pull request by joachifm**
###### Motivation for this change

Updating the resolver list manually is a waste of time.
###### Things done

Tested in a VM. Works as expected.

---

This removes the abillity to specify a resolver list.  Instead, we always use the upstream resolver list.  This list is then updated (and verified) periodically, to ensure that we're always using a valid server list.

### CODE REVIEW

File: nixos/modules/services/networking/dnscrypt-proxy.nix [hunk 1]
```diff
@@ -191,6 +197,62 @@ in
     };
     users.groups.dnscrypt-proxy = {};
 
+    systemd.services.init-dnscrypt-proxy-datadir = {
+      description = "Initialize dnscrypt-proxy data directory";
+
+      after = [ "local-fs.target" ];
+
+      script = ''
+        mkdir -p ${dataDir}
```
Comment (joachifm) [inline nixos/modules/services/networking/dnscrypt-proxy.nix]: Need `unitConfig.RequiresMountsFor = [ dataDir ]`

File: nixos/modules/services/networking/dnscrypt-proxy.nix [hunk 2]
```diff
@@ -178,7 +178,7 @@ in
 
         ${resolverList} r,
       }
-    ''));
+    '')); # */
```
Comment (Mic92) [inline nixos/modules/services/networking/dnscrypt-proxy.nix]: what is this?

Comment (joachifm) [inline nixos/modules/services/networking/dnscrypt-proxy.nix]: Just a temporary hack to "fix" emacs syntax highlighting.

**Comment by spacekitteh** [comment]
@grahamc security

**Pull request merged**